### PR TITLE
Throw exception when ufs journal reads unrecognized entry

### LINKED ISSUE #9751: UfsJournal ignores unrecognized journal entries
**Alluxio Version:**
2.1.0-SNAPSHOT

**Describe the bug**
An unrecognized journal entry is simply ignored.

**To Reproduce**
- Write a journal entry newly added in v2.x
- Read journal from v2.y < v2.x
- Master starts up successfully

**Expected behavior**
Master should crash with `IllegalStateException` as a journal entry was unrecognized

**Urgency**
Medium, this an an early detection of journal compatibility 

**Additional context**
`EmbeddedJournal` calls `JournalAssociation` and will throw an `IllegalStateException` whereas `UfsJournal` does not

### PR DESCRIPTION
**Pull request by gpang**
fixes #9751

### CODE REVIEW

File: core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournal.java [hunk 1]
```diff
@@ -415,7 +415,11 @@ private long catchUp(JournalReader journalReader) {
           case LOG:
             JournalEntry entry = journalReader.getEntry();
             try {
-              mMaster.processJournalEntry(entry);
+              if (!mMaster.processJournalEntry(entry)) {
+                JournalUtils
+                    .handleJournalReplayFailure(LOG, null, "%s: Unrecognized journal entry: %s",
+                        mMaster.getName(), entry);
+              }
               JournalUtils.sinkAppend(mJournalSinks, entry);
```
Comment (apc999) [inline core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournal.java]: should this be in `else` branch?

**Comment by apc999** [comment]
one question left. Otherwise LGTM

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Comment by apc999** [comment]
alluxio-bot, cherry-pick this to branch-2.0 please

**Pull request merged**